persistence/soft-index/src/main/java/org/infinispan/persistence/sifs/EntryHeader.java

```diff
@@ -53,12 +60,26 @@ public long expiryTime() {
       return expiration;
    }
 
+   public int getHeaderLength() {
+      return headerLength;
+   }
+
    @Override
    public String toString() {
       return String.format("[keyLength=%d, valueLength=%d, metadataLength=%d, internalMetadataLength=%d,seqId=%d, expiration=%d]", keyLength, valueLength, metadataLength, internalMetadataLength, seqId, expiration);
    }
 
    public int totalLength() {
-      return keyLength + metadataLength + internalMetadataLength + valueLength + HEADER_SIZE;
+      return keyLength + metadataLength + internalMetadataLength + valueLength + headerLength;
+   }
+
+   public static void writeHeader(ByteBuffer buf, short keyLength, short metadataLength, int valueLength, int internalMetadataLength, long seqId, long expiration) {
+      buf.putInt(EntryHeader.MAGIC);
```

Comment:
I would prefer to avoid writing MAGIC per header if possilbe. However, if we can't avoid that we should at least store it as a single byte. `useMagic` was only ever used for testing, so we don't have to worry about the compatibility of this field as it was never used in production.

Reply:
I'll change to byte. 
IMO, it is better to keep a `magic` number per entry. If I understand correctly, SIFS only appends the keys changes to a new file; So, if the format changes in the future, we don't need any migration during startup.